runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/util/Stager.java

```diff
@@ -20,10 +20,32 @@
 import com.google.api.services.dataflow.model.DataflowPackage;
 import java.util.List;
 
-/**
- * Interface for staging files needed for running a Dataflow pipeline.
- */
+/** Interface for staging files needed for running a Dataflow pipeline. */
 public interface Stager {
-  /* Stage files and return a list of packages. */
-  List<DataflowPackage> stageFiles();
+  /**
+   * Stage default files and return a list of {@link DataflowPackage} objects describing the actual
+   * location at which each file was staged.
+   *
+   * <p>This is required to be identical to calling {@link #stageFiles(List)} with the default set
+   * of files.
+   *
+   * <p>The default is controlled by the implementation of {@link Stager}. The only known
+   * implementation of stager is {@link GcsStager}. See that class for more detail.
+   */
+  List<DataflowPackage> stageDefaultFiles();
+
+  /**
+   * Stage files and return a list of packages {@link DataflowPackage} objects describing th actual
+   * location at which each file was staged.
+   *
+   * <p>The mechanism for staging is owned by the implementation. The only requirement is that the
+   * location specified in the returned {@link DataflowPackage} should, in fact, contain the
+   * contents of the staged file.
+   */
+  List<DataflowPackage> stageFiles(List<String> filesToStage);
+
+  /**
+   * Stage bytes to a target file name wherever this stager stages things.
```

Comment:
"target" implies the full output path instead of just a local name for the bytes which the stager can use to place somewhere. Consider using a different name then "target"

Reply:
Done. Tried `baseName` on for size.